src/wp-includes/option.php

```diff
@@ -1133,6 +1140,65 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ *  Determines the appropriate autoload value for an option based on input.
+ *
+ *  This function checks the provided autoload value and returns a standardized
+ *  value ('yes', 'no', 'default-yes', or 'default-no') based on specific conditions.
+ *
+ * If no valid autoload value is provided, the function will check the size of the given option value
+ * and compare it to the maximum allowed size. If it ls larger, it will return 'default-no', otherwise
+ * it will return 'default-yes'.
+ *
+ * @since 6.5.0
+ * @access private
+ *
+ * @param string           $option   The name of the option.
+ * @param mixed            $value    The value of the option to check its autoload value.
+ * @param string|bool|null $autoload The autoload value to check.
+ *                                   Accepts 'yes'|true to enable or 'no'|false to disable, or
+ *                                   'default-yes' and 'default-no' for internal purposes.
+ *                                   Any other autoload value will be forced to either `default-yes`
+ *                                   or `default-no`.
+ * @return string Returns the original $autoload value if valid, or 'default-no' if the size exceeds
+ *                the maximum allowed size, or 'default-yes' otherwise.
+ */
+function determine_option_autoload_value( $option, $value, $autoload ) {
+
+	// Check if autoload is a boolean.
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'yes' : 'no';
+	}
+
+	switch ( $autoload ) {
+		case 'no':
+		case 'yes':
+		case 'default-yes':
+		case 'default-no':
+			return $autoload;
+
+	}
```

Comment:
Doesn't name sense to have this imho. Makes it much harder to understand and use.

Instead there should be a filter that runs per option (adding or updating) that would let plugins override the automated decision whether to autoload or not. The filter to set the cut-off side is okay, but is not enough.

Reply:
The specific logic here is purely to validate the input value, to ensure no other strings can be provided (which wouldn't work anyway).

I agree with you that a filter to override the core decision could be helpful. FWIW though that can already be accomplished even with the currently proposed `wp_max_autoloaded_option_size` filter. For example:
```php
add_filter(
	'wp_max_autoloaded_option_size',
	function ( $max_size, $option ) {
		if ( 'my_option' === $option ) {
			return 0; // Forces to never autoload.
			// Alternatively return PHP_INT_MAX; to always autoload.
		}
		return $max_size;
	},
	10,
	2
);
```

That said, that behavior is not very intuitive, so another filter to make the decision could be helpful.

Reply:
added new filter option_autoload_value
We could make(or add as well) the dynamic filter like option_autoload_value_{ $option }